model/build/build.go

```diff
@@ -247,6 +247,16 @@ func (b *Build) MarkFinished(status string, finishTime time.Time) error {
 	)
 }
 
+func (b *Build) GetTimeSpent() (time.Duration, time.Duration, error) {
+	tasks, err := task.FindAllFirstExecution(task.ByBuildId(b.Id).WithFields(task.TimeTakenKey, task.StartTimeKey, task.FinishTimeKey, task.DisplayOnlyKey, task.ExecutionKey))
```

Comment:
Is first execution what we want? I'd think we want the latest execution for each task.

Reply:
The thinking of using the first execution is that when a task gets restarted the build's makespan shouldn't be the time from the start of the first task to run to the finish of the restarted task (potentially days).
This was the solution I landed on in an earlier PR (#3955).

Reply:
Oh I see, that makes sense.